tests/Tests/500__jupyterhub/jupyterhub-user-access.robot

```diff
@@ -0,0 +1,89 @@
+*** Settings ***
+Force Tags       Smoke  Sanity
+Resource         ../../Resources/ODS.robot
+Resource         ../../Resources/Common.robot
+Resource         ../../Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot
+Library          DebugLibrary
+Library          JupyterLibrary
+Library          OpenShiftCLI
+Library          OperatingSystem
+Suite Setup      Special User Testing Suite Setup
+Suite Teardown   End Web Test
+
+*** Variables ***
+
+*** Test Cases ***
+Test User Not In JH Access Groups
+    [Tags]  Smoke
+    ...     PLACEHOLDER  #Category tags
+    ...     ODS-503
+    Open Browser  ${ODH_DASHBOARD_URL}  browser=${BROWSER.NAME}  options=${BROWSER.OPTIONS}
+    Login To RHODS Dashboard  ${TEST_USER.USERNAME}  ${TEST_USER.PASSWORD}  ${TEST_USER.AUTH_TYPE}
+    Wait for RHODS Dashboard to Load
+    Launch JupyterHub From RHODS Dashboard Dropdown
+    Login Verify Not Allowed  ${TEST_USER.USERNAME}  ${TEST_USER.PASSWORD}  ${TEST_USER.AUTH_TYPE}
+
+Test User In JH Admin Group
+    [Tags]  Smoke
+    ...     PLACEHOLDER  #Category tags
+    ...     ODS-503
+    Open Browser  ${ODH_DASHBOARD_URL}  browser=${BROWSER.NAME}  options=${BROWSER.OPTIONS}
+    Login To RHODS Dashboard  ${TEST_USER.USERNAME}  ${TEST_USER.PASSWORD}  ${TEST_USER.AUTH_TYPE}
+    Wait for RHODS Dashboard to Load
+    Launch JupyterHub From RHODS Dashboard Dropdown
+    Login Verify Admin Allowed  ${TEST_USER.USERNAME}  ${TEST_USER.PASSWORD}  ${TEST_USER.AUTH_TYPE}
+
+Test User In JH Users Group
+    [Tags]  Smoke
+    ...     PLACEHOLDER  #Category tags
+    ...     ODS-503
+    Open Browser  ${ODH_DASHBOARD_URL}  browser=${BROWSER.NAME}  options=${BROWSER.OPTIONS}
+    Login To RHODS Dashboard  ${TEST_USER.USERNAME}  ${TEST_USER.PASSWORD}  ${TEST_USER.AUTH_TYPE}
+    Wait for RHODS Dashboard to Load
+    Launch JupyterHub From RHODS Dashboard Dropdown
+    Login Verify Regular User Allowed  ${TEST_USER.USERNAME}  ${TEST_USER.PASSWORD}  ${TEST_USER.AUTH_TYPE}
+
+
+*** Keywords ***
+Special User Testing Suite Setup
+  Set Library Search Order  SeleniumLibrary
```

Comment:
I don't think this needs to be in the test suite setup or in its own keyword. You could just use it in the first test case and it should keep the search order for the whole test suite (i.e. this .robot file)
https://robotframework.org/robotframework/latest/libraries/BuiltIn.html#Set%20Library%20Search%20Order

Reply:
I prefer @crobby's approach (library search order configured in Suite Setup) because tests should be able to run independently of each others. You don't have guarantee that the other test will be run before.

Reply:
IMHO the keywords should be explicitly called with the `library.keyword` format to avoid any possible confusion/errors rathen than setting the search order

Reply:
I agree. The problem is that a big part of that work should be done in the Resources folder and that's out of the scope of this PR

Reply:
From my point of view, if you use just one or two libraries is ok to use the long name to avoid conflicts,  but as the project grows we will probably end up using more libraries or resource files.  The user guide of Robot Framework says that a solution to keyword conflict is to set the library order. This is from the part called "Specifying explicit priority between libraries and resources" of their user guide https://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html